### PR TITLE
TST: Verify isin returns consistent results for passed iterables

### LINKED ISSUE #50234: BUG: pd.Series.isin returns inconsistent results for a passed iterable depending on the dtype
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

df = pd.DataFrame({"int": [1, 2, 3], "float": [1.0, 2.0, 3.0]})
filter_ = [1, 2]

print('## Good')
print(df["int"].isin([1, 2]))
print(df["float"].isin([1, 2]))
print(df["int"].isin(i for i in [1.0, 2.0]))
print('## Bad')
print(df["float"].isin(i for i in [1.0, 2.0]))
```

### Issue Description

Passing an iterator to `pd.Series.isin` does not seem to work reliably. Depending on the dtype (I checked `int` and float`), the behavior differs.

The `int` dtype filters as I would expect (even when passing floats) while the float dtype always returns `False`.

I was running into this issue in production where all but one column were ints.

### Expected Behavior

The output is always 

```
0     True
1     True
2    False
```

Or at least consistent independent of the type.

Instead the last case returns

```
0    False
1    False
2    False
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 8dab54d6573f7186ff0c3b6364d5e4dd635ff3e7
python           : 3.10.5.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 22.1.0
Version          : Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:54 PDT 2022; root:xnu-8792.41.9~2/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.UTF-8

pandas           : 1.5.2
numpy            : 1.23.5
pytz             : 2022.6
dateutil         : 2.8.2
setuptools       : 65.6.3
pip              : 22.3.1
Cython           : None
pytest           : 7.2.0
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.9.1
html5lib         : 1.1
pymysql          : None
psycopg2         : 2.9.5
jinja2           : 3.1.2
IPython          : 8.6.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : 1.0.9
fastparquet      : None
fsspec           : 2022.11.0
gcsfs            : 2022.11.0
matplotlib       : 3.6.2
numba            : None
numexpr          : 2.8.4
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : None
pyarrow          : 10.0.1
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.9.3
snappy           : None
sqlalchemy       : 1.4.44
tables           : None
tabulate         : 0.9.0
xarray           : 0.21.1
xlrd             : 2.0.1
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by jessica-writes-code**
Add tests to verify that the `Series` method `isin` returns consistent results when passed iterables with different dtypes. (See #50234.)

- [x] closes #50234
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] ~Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.~ (Not applicable.)
- [x] ~Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.~ (Not applicable.)

### CODE REVIEW

File: pandas/tests/series/methods/test_isin.py [hunk 1]
```diff
@@ -234,3 +234,28 @@ def test_isin_filtering_with_mixed_object_types(data, is_in):
     expected = Series([True, False])
 
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "data,is_in,expected",
```
Comment (MarcoGorelli) [inline pandas/tests/series/methods/test_isin.py:240]: looks like `is_in` and `expected` are the same for the two test cases - if so, perhaps just define them in the test body, and only parametrize over `data`?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
This is off to a good start!

File: pandas/tests/series/methods/test_isin.py [hunk 2]
```diff
@@ -234,3 +234,17 @@ def test_isin_filtering_with_mixed_object_types(data, is_in):
     expected = Series([True, False])
 
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [([1, 2, 3]), ([1.0, 2.0, 3.0])],
```
Comment (MarcoGorelli) [inline pandas/tests/series/methods/test_isin.py:241]: looks like you've got some extra brackets (I'm surprised none of the linters caught this)
```suggestion
    [[1, 2, 3], [1.0, 2.0, 3.0]],
```

Comment (MarcoGorelli) [inline pandas/tests/series/methods/test_isin.py:241]: looks like you've got some extra brackets (I'm surprised none of the linters caught this)
```suggestion
    [[1, 2, 3], [1.0, 2.0, 3.0]],
```

File: pandas/tests/series/methods/test_isin.py [hunk 3]
```diff
@@ -234,3 +234,17 @@ def test_isin_filtering_with_mixed_object_types(data, is_in):
...
+)
+def test_isin_filtering_on_iterable(data):
+    # GH 50234
+
+    ser = Series(data)
+    result = ser.isin(i for i in [1, 2])
```
Comment (MarcoGorelli) [inline pandas/tests/series/methods/test_isin.py:247]: sorry, I didn't catch this the first time round - doesn't the original issue have `(i for i in [1.0, 2.0])`?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
couple more minor comments, then looks good to me

@phofl any further comments?

File: pandas/tests/series/methods/test_isin.py (hunk 3 above)
Comment (phofl) [reply pandas/tests/series/methods/test_isin.py:247]: Agreed, we should parametrize over both, e.g. [1, 2] and [1.0, 2.0]

File: pandas/tests/series/methods/test_isin.py (hunk 2 above)
Comment (jessica-writes-code) [reply pandas/tests/series/methods/test_isin.py]: Good catch! Thanks!

File: pandas/tests/series/methods/test_isin.py (hunk 3 above)
Comment (jessica-writes-code) [reply pandas/tests/series/methods/test_isin.py]: Good point! Updated parameters accordingly.

**Review by MarcoGorelli — APPROVED** [review]
Nice, thanks @jessica-writes-code ! Looks good to me pending green

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
Thx @jessica-writes-code

**Pull request merged**